lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -209,9 +227,35 @@ def get(self, request, *args, **kwargs):
             'user_timezone': user_timezone,
         }
 
+        # Only show the set course goal message for enrolled, unverified
+        # users in a course that allows for verified statuses.
+        is_already_verified = CourseEnrollment.is_enrolled_as_verified(request.user, course_key)
+        if has_course_goal_permission(request, course_key_string, {'is_enrolled': is_enrolled}) \
+                and not is_already_verified:
+            goal_options = get_course_goal_options()
+            ordered_goal_options = self.sort_course_goals(goal_options)
+            course_goals = {
+                'goal_options': ordered_goal_options,
+                'selected_goal': None
+            }
+
+            selected_goal = get_course_goal(request.user, course_key)
+            if selected_goal:
+                goal_key = selected_goal.goal_key
+                course_goals['selected_goal'] = {
+                    'key': goal_key,
+                    'text': goal_options[goal_key]
+                }
```

Comment:
Yeah I'm not sure why I separated goalOptions into `[{'key': ..., 'text': ...},...,{}]`. I'm refactoring this into `{key1: text1, key2: text2}` because it's cleaner in both the API and MFE. This will also remove the need `sort_course_goals`.